### ISSUE TITLE
@stylistic/object-property-newline: more flexibility?

### ISSUE THREAD

**Issue by GBrachetta** [eslint-stylistic/eslint-stylistic#234]
### Clear and concise description of the problem

The rule `@stylistic/object-property-newline` could have more flexibility, like `@stylistic/object-curly-newline` has, so it could allow to have the following result for **imports**:

```js
import {
  Loader2,
  MSquareIcon,
  RadarIcon,
  Table,
  Trash,
} from "lucide-react";
```

While keeping the one-line format for other cases, such as:

```js
export const people = [
  { age: 29, gender: "male", name: "Jane", surname: "Doe" },
  { age: 30, gender: "male", name: "Bob", surname: "Doe" },
  { age: 27, gender: "male", name: "Jill", surname: "Doe" },
  { age: 26, gender: "male", name: "Jake", surname: "Doe" },
];
```

### Suggested solution

It would be great if that rule (`@stylistic/object-property-newline`) could receive configuration options such as `@stylistic/object-curly-newline`, so import declaration and other cases could be configured differently.

### Alternative

Additionally, instead of having an item count to trigger the rule, it would be great if line length would be an option, so properties could be wrapped if the line excedes for example 80 characters.

### Validations

- [X] Follow our [Code of Conduct](https://github.com/eslint-stylistic/eslint-stylistic/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guide](https://eslint.style/contribute/guide).
- [X] Check that there isn't already an issue that request the same feature to avoid creating a duplicate.

### Contributes

- [ ] If this feature request is accepted, I am willing to submit a PR to fix this issue

**Comment by antfu** [comment]
Sounds reasonable to have options for it. What's the options design in your mind?

**Comment by hackloud** [comment]
```typescript
/* object-curly-newline: [ 'error', { ImportDeclaration: { allowAllPropertiesOnSameLine: false, minProperties: 2 } } ] */

// Disallow
import { foo, bar } from 'module';

// Allow
import {
  foo,
  bar,
} from 'module';

// Also allow another expression
const { foo, bar } = someObject;
```

I'd like to enforce `object-property-newline`only for ImportDeclaration like above.
I think same options like [object-curly-newline](https://eslint.style/rules/js/object-curly-newline) are good.
(ObjectExpression, ObjectPattern, ImportDeclaration, ExportDeclaration / minProperties)

**Comment by s-hoff** [comment]
I'd like to see the same options as `object-curly-newline` as well.

Said rule is great and has the right options, but doesn't add line breaks between properties, only around the curlies.
Not even just for imports, I currently cannot even find a way to force an object with, say, at least three properties to go multiline.

```js
'@stylistic/object-curly-newline': ['error', {
  minProperties: 3,
  multiline: true,
}],
```
Turns
```js
const sameline = { foo: 'lish', bar: 'line' };
const multiline = { foo: 'lish', bar: 'line', break: 'here' };
```
into
```js
const sameline = { foo: 'lish', bar: 'line' };
const multiline = {
  foo: 'lish', bar: 'line', break: 'here',
};
```

If I try to add `object-property-newline` to also get the line breaks between the multiline version, it wants to multiline the `sameline` version as well.
```js
const sameline = { foo: 'lish', bar: 'line' }; // <- Object properties must go on a new line.  ([object-property-newline](https://eslint.org/docs/rules/object-property-newline))
const multiline = {
  foo: 'lish', bar: 'line', break: 'here', // of course also errors but here its what we want
};
```

Following all the autofixes this would turn into
```js
// two is supposed to be allowed as sameline by minProperties: 3
const sameline = {
  foo: 'lish',
  bar: 'line',
};
const multiline = {
  foo: 'lish',
  bar: 'line',
  break: 'here',
};
```

As it looks like the two are meant to be used together, they should also share the same configuration options. As it is now, using them together limits the effective options for `object-curly-newlines` as well.

**Comment by AlbertMarashi** [comment]
> I'd like to enforce object-property-newlineonly for ImportDeclaration like above.

Please i'm begging for this 😆

**Comment by SoCuul** [comment]
This would be very nice to have, as I use the "minItems" option on "array-element-newline" and this would make it possible to do the equivalent for objects.

**Comment by HyperLife1119** [comment]
I also want this rule to apply to import/export statements.
